### ISSUE TITLE
Can't load Advance Wars .sav files + graphical glitches when saving

### ISSUE THREAD

**Issue by ghost** [libretro/gpsp#15]
I'm using the 3ds port of retroarch, and while it seems that gpSP can create .sav files, (although saving during a game will result in graphical glitches, such as river tiles turning black at regular intervals, until the next battle between units) the core isn't able to load them the next time I start the game.
I will try to upload screenshot of the glitches if that can be of use.

**Comment by AEFeinstein** [comment]
I'm using the Raspberry Pi version of retroarch and am experiencing the same .sav issue. I can't vouch for the graphical glitches. mGBA manages to save and load the .sav files just fine, but is painfully slow.

**Comment by th4t9uy** [comment]
I have a similar issue with Advance Wars. I save my progress in game and I can see that a .sav file is created but when I start Advance Wars again my saved progress isn't loaded and I have to start again. 

From what I can tell the issue seems to be limited to Advance Wars, other games can load my saved progress just fine.

**Comment by AEFeinstein** [comment]
I did a little poking around and I was able to replicate this bug on the x86 build of libretro/gpsp. I was *not* able to replicate this bug on the x86 build of notaz/gpsp, so something changed between the two which broke saving the game. The projects have diverged enough that diffing didn't yield any solutions.

@twinaphex any tips on debugging this? I'm a decent C programmer, but have never dug into an emulator before.
